### DISCUSSION TITLE
`$embed_file` should always embed "for real" when cross-compiling.

### DISCUSSION THREAD

**Discussion by JalonSolov** [vlang/v#14666] [Ideas]
Currently, `$embed_file` will load the file at runtime if compiling without `-prod`, but with `-prod`, it really embeds the file.

However, when cross-compiling, the file really needs to be embedded even without `-prod`, since file paths rarely match across systems.

It should be relatively simple to fix - for someone who can find the right flags.

Line 804 in cgen.v has
```v
		if g.embed_file_is_prod_mode() {
```
that could be changed to something like
```v
		if g.embed_file_is_prod_mode() || target != current_target || os != current_os {
```

For now, the only "workaround" is to always compile with `-prod` when cross-compiling, but that is suboptimal for the development cycle.

**Comment by Wertzui123** [comment]
I think I'd genereally always make `$embed_file` really embed the file, as it's hard to debug it instead and it just feels weird. Or at least there should be an option like `-always-embed`.

**Comment by larpon** [comment]
There currently is a flag (that need a rename IMO):
`-d debug_embed_file_in_prod`

**Comment by JalonSolov** [comment]
It would be better as `-d force_embed_file`.  Whatever name is used, this is another workaround, but it shouldn't be required for cross-compiles.
